### PR TITLE
Data Directory support

### PR DESCRIPTION
**Pull request by erik-ilyin**
forEach method
create, delete function

### CODE REVIEW

File: Algorithmia/Data/AlgoDataDirectory.swift [hunk 1]
```diff
@@ -8,21 +8,183 @@
...
+        super.init(client: client, dataUrl: dirUrl, type: .Directory)
+    }
+    
+    func create(_ completion:@escaping AlgoSimpleCompletionHandler) {
+        let entity = try! AlgoJSONEntity(entity: ["name":self.basename()])
+        _ = client.send(method: .POST, path: self.parent()!.getUrl(), data: entity) { (respData, error) in
+            if respData.statusCode == 200 {
+                completion(nil)
+            }
+            else if error == nil {
```
Comment (anowell) [inline Algorithmia/Data/AlgoDataDirectory.swift:39]: @pmcq - I know algorithms can fail and return 200, but do we ever return 200 if data operations fail?

Comment (pmcq) [reply Algorithmia/Data/AlgoDataDirectory.swift:39]: So... there's one place that might happen, which is awkward and I don't think should necessarily be thrown as an exception (it isn't in any other client I don't think):

Deleting a directory you might delete N files but due to some error you can't delete M files:
http://docs.algorithmia.com/#deleting-a-directory

It's... kind of awful

Comment (anowell) [reply Algorithmia/Data/AlgoDataDirectory.swift:39]: Yikes. ok. So if I follow correctly, if data requests are 200, then clients should consider it a success.

(And IMO, we should consider having partially failed deletes return a non-200 response.)

Comment (pmcq) [reply Algorithmia/Data/AlgoDataDirectory.swift:39]: Yup that's correct, and I also think we should change the partially failed delete to return non-200 for sanity.

**Comment by anowell** [comment]
looks good to me.

**Pull request merged**